src/MessageDispatch.KurrentDB/KurrentDbSubscriber.cs

```diff
@@ -260,21 +262,56 @@ public async void Start()
                 IsLive = false;
                 _logger.LogInformation(ex, "Event Store subscription dropped {0}", SubscriptionDroppedReason.Disposed);
                 break;
-            }
+            }
+#if NETFRAMEWORK
+            catch (Grpc.Core.RpcException ex)
+            {
+                var innerWebException = ex.InnerException.InnerException;
+                if (innerWebException != null && innerWebException.Message.Contains("Error 12002"))
+                {
+                    _logger.LogInformation(ex, "Event Store subscription dropped {0}", SubscriptionDroppedReason.SubscriberError);
+                    TryCreatingSubscription();
+                }
+            }
+#endif
             catch (Exception ex)
             {
                 IsLive = false;
                 _startingPosition = _lastProcessedEventPosition;
                 _logger.LogError(ex, "Event Store subscription dropped {0}", SubscriptionDroppedReason.SubscriberError);
                 Console.WriteLine(ex);
-            }
-
-            // Sleep between reconnections to not flood the database or not kill the CPU with infinite loop
-            // Randomness added to reduce the chance of multiple subscriptions trying to reconnect at the same time
+            }
+
+            // Sleep between reconnections to not flood the database or not kill the CPU with infinite loop
+            // Randomness added to reduce the chance of multiple subscriptions trying to reconnect at the same time
             await Task.Delay(1000 + new Random((int)DateTime.UtcNow.Ticks).Next(1000));
         }
-    }
-
+    }
+
+#if NETFRAMEWORK
+    private void TryCreatingSubscription()

```

Comment:
I suspect this could probably all be cleaned up by just converting to something like an ImmediateRetry flag inside `Start()`

- If we handle our special exception, set it to true, don't touch IsLive -> resubscribe. EXCEPT if it is already true, in which case we'd follow the normal Exception logic because we do not do 2 immediate retries in a row.
- Other exceptions ImmediateRetry = false, proceed as normal.
- If immediateRetry = true, skip the delay at the end of the loop
- Set immediateRetry back to false if we successfully subscribe.

This means we just use the existing resubscribe loop and don't need the awkwardness that extracting this adds (having to store the subscription in a field etc.)